lib/src/bitonic_sort.dart

```diff
@@ -0,0 +1,181 @@
+  /// of [direction].
+  _BitonicMerge(
+      Logic clk, Logic reset, List<Logic> bitonicSequence, int direction)
+      : super(name: 'bitonic_merge') {
+    clk = addInput('clk', clk);
+    reset = addInput('reset', reset);
+
+    for (var i = 0; i < bitonicSequence.length; i++) {
+      _inputs.add(
+          addInput('x$i', bitonicSequence[i], width: bitonicSequence[i].width));
+    }
+
+    if (_inputs.length > 1) {
+      final k = _inputs.length ~/ 2;
+
+      for (var i = 0; i < 0 + k; i++) {
+        final indexA = i;
+        final indexB = i + k;
+        final swap =
+            _CompareSwap(clk, reset, _inputs, indexA, indexB, direction);
+        _inputs = swap.swapped;
+      }
+
+      for (var i = 0; i < _inputs.length; i++) {
+        _outputs.add(addOutput('y$i', width: _inputs[i].width));
+        _outputs[i] <= _inputs[i];
+      }
+
+      final mergeLeft = _BitonicMerge(clk, reset,
+          _inputs.getRange(0, _inputs.length ~/ 2).toList(), direction);
+      final mergeRight = _BitonicMerge(
+          clk,
+          reset,
+          _inputs.getRange(_inputs.length ~/ 2, _inputs.length).toList(),
+          direction);
+
+      final mergeRes = mergeLeft.sorted + mergeRight.sorted;
+
+      for (var i = 0; i < mergeRes.length; i++) {
+        _outputsFinal.add(addOutput('sorted_$i', width: mergeRes[i].width));
+        _outputsFinal[i] <= mergeRes[i];
+      }
+    } else if (_inputs.length == 1) {
+      _outputsFinal.add(addOutput('sorted_0', width: _inputs[0].width));
+      _outputsFinal[0] <= _inputs[0];
+    }
+  }
+}
+
+/// Sort [inputs] to specified order.
+class BitonicSort extends Module {
+  /// The list of inputs port.
+  final List<Logic> _inputs = [];
+
+  /// The list of outputs port.
+  final List<Logic> _outputs = [];
+
+  /// The [sorted] result.
+  List<Logic> get sorted => _outputs;
+
+  /// Constructs a [Module] to sort list of [Logic] based on [direction].
+  ///
+  /// The sorting module will recursively split inputs into a bitonic sequence
+  /// and passed to [_BitonicMerge] to perform merging process and return
+  /// the final [sorted] results.
+  BitonicSort(Logic clk, Logic reset, List<Logic> inputs, int direction,
+      {super.name}) {
+    clk = addInput('clk', clk);
+    reset = addInput('reset', reset);
+
+    for (var i = 0; i < inputs.length; i++) {
+      _inputs.add(addInput('x$i', inputs[i], width: inputs[i].width));
+    }
+
+    if (_inputs.length > 1) {
+      final k = _inputs.length ~/ 2;
+      final sortLeft = BitonicSort(
+          clk, reset, _inputs.getRange(0, _inputs.length ~/ 2).toList(), 1,
+          name: 'sort_left_$k');
+
+      final sortRight = BitonicSort(clk, reset,
+          _inputs.getRange(_inputs.length ~/ 2, _inputs.length).toList(), 0,
+          name: 'sort_right_$k');
+
+      final res = sortLeft.sorted + sortRight.sorted;
+
+      final y = _BitonicMerge(clk, reset, res, direction);
```

Comment:
Why not make this something like
```dart
final mergeResult = _BitonicMerge(clk, reset, res, direction).sorted;
```
Then we don't have this poorly-named `y` variable and also don't have to keep calling `.sorted` below.